### PR TITLE
bug 1622822. Add uuid to acl docs

### PR DESCRIPTION
**Pull request by jcantrill**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1622822
* adds uid to permissions
* adds uid to index-pattern

### CODE REVIEW

**Comment by jcantrill** [comment]
yes.  all ACLs are supposed to expire every 60s

**Comment by jcantrill** [comment]
@ewolinetz @richm updated to include modifications to index pattern to include uid

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaUtils.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            return Project.EMPTY;
+        }
+
+        Matcher matcher = reIndexPattern.matcher(index);
+        if (matcher.matches()) {
+            return new Project(matcher.group("name"), matcher.group("uid"));
+        }
+        return new Project(index, null);
+    }
+
```
Comment (richm) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaUtils.java]: if the index is `project.*`, then index = `project.*` and uid = `null`
then `formatIndexPattern` will return `project.project.*.*`?

File: src/test/resources/io/fabric8/elasticsearch/plugin/project_strategy_roles_shared_non_ops_kibana_mode.yml [hunk 2]
```diff
@@ -11,9 +11,9 @@ gen_ocp_kibana_shared_non_ops:
 gen_project_foo_bar:
   expires: '10'
   indices:
-    ?project?foo?bar?*:
+    ?project?foo?bar?123abc?*:
```
Comment (richm) [inline src/test/resources/io/fabric8/elasticsearch/plugin/project_strategy_roles_shared_non_ops_kibana_mode.yml]: in this case - `project` is the project prefix, `foo` is the namespace name?  what is `bar`?  is `123abc` the uuid?

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaUtils.java (hunk 1 above)
Comment (richm) [reply src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaUtils.java]: ok - hard to tell from the flow - but if you are sure that this method is never called with "project.*" then that's good enough for me

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**